src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -58,184 +61,336 @@ using namespace chip;
+                              "Network ID must be larger than Thread extended PAN ID!");
+                if ((error = dataset.GetExtendedPanId(extendedPanId)) != CHIP_NO_ERROR)
+                {
+                    ChipLogDetail(Zcl, "Failed to get Thread extpanid: %s", ErrorStr(error));
+                    err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_UNKNOWN_ERROR;
+                    break;
+                }
+                memcpy(sNetworks[i].mNetworkID, extendedPanId, sizeof(extendedPanId));
+                sNetworks[i].mNetworkIDLen = sizeof(extendedPanId);
+
+                sNetworks[i].mNetworkType = NetworkType::kThread;
+                sNetworks[i].mEnabled     = false;
+
+                err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_SUCCESS;
+                break;
+            }
+        }
+    }
 #else
-    // The target does not supports ThreadNetwork. We should not add AddThreadNetwork command in that case then the upper layer will
-    // return "Command not found" error.
-    return EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_UNKNOWN_ERROR;
+    err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
 #endif
+
+    TLV::TLVWriter * writer          = nullptr;
+    app::CommandPathParams cmdParams = { /* endpoint id */ endpointId, /* group id */ 0, ZCL_NETWORK_COMMISSIONING_CLUSTER_ID,
+                                         ZCL_ADD_THREAD_NETWORK_RESPONSE_COMMAND_ID,
+                                         (chip::app::CommandPathFlags::kEndpointIdValid) };
+    SuccessOrExit(encodingError = apCommandObj->PrepareCommand(&cmdParams));
+    VerifyOrExit((writer = apCommandObj->GetCommandDataElementTLVWriter()) != nullptr, encodingError = CHIP_ERROR_INCORRECT_STATE);
+    SuccessOrExit(encodingError = writer->Put(TLV::ContextTag(0), static_cast<uint32_t>(err)));
+    SuccessOrExit(encodingError = writer->PutString(TLV::ContextTag(1), ""));
+    SuccessOrExit(encodingError = apCommandObj->FinishCommand());
+exit:
+    if (encodingError != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to encode response command for AddThreadNetwork: %d (Command Error: %d)", encodingError, err);
+    }
 }
 
-EmberAfNetworkCommissioningError OnAddWiFiNetworkCommandCallbackInternal(app::Command *, EndpointId, ByteSpan ssid,
-                                                                         ByteSpan credentials, uint64_t breadcrumb,
-                                                                         uint32_t timeoutMs)
+void OnAddWiFiNetworkCommandCallbackInternal(app::Command * apCommandObj, EndpointId endpointId, ByteSpan ssid,
+                                             ByteSpan credentials, uint64_t breadcrumb, uint32_t timeoutMs)
 {
-#if defined(CHIP_DEVICE_LAYER_TARGET)
+    CHIP_ERROR encodingError             = CHIP_NO_ERROR;
     EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
-
-    for (size_t i = 0; i < kMaxNetworks; i++)
+#if defined(CHIP_DEVICE_LAYER_TARGET)
```

Comment:
Again, if WiFi is only supported when CHIP_DEVICE_LAYER_TARGET, why does this function exist in any other cases?